### ISSUE TITLE
Reactivity issue when passing a signal

### ISSUE THREAD

**Issue by OrJDev** [solidjs-community/eslint-plugin-solid#54]
So i have this library https://github.com/OrJDev/solid-trpc, a function called useQuery:

Use query should take a signal,

const [name, setName] = createSignal("john");

And should be used like

useQuery(name)

Or

useQuery(()=> name().toUpperCase())

However i keep getting a reactivity warning, how come? This seems like a perfectly valid usecase

**Comment by OrJDev** [comment]
You can repro in

https://github.com/OrJDev/voting-jd

 ```ts
// eslint-disable-next-line solid/reactivity
  const questions = trpc.question.getSome.useQuery(() => page());
```

**Comment by joshwilsonvu** [comment]
I agree, it shouldn't warn here--looks like the same issue as in #50. I'll update you when that's resolved.

**Comment by OrJDev** [comment]
Looks like this was partially fixed

```ts
const addRes = add(
    () => ({
      a: num1(),
      b: 3,
    }),
    () => ({
      placeholderData: (prev) => prev,
      onSettled(data, error) {
        console.log({ data, error })
      },
      onSuccess: (data) => {
        console.log('success', data)
      },
      alwaysCSRRedirect: true,
    })
  )
```

This seems to throw the exact same issue
